crossbar/twisted/endpoint.py

```diff
@@ -213,6 +213,12 @@ def _create_tls_server_context(config, cbdir, log):
 
     # create a TLS context factory
     # see: https://twistedmatrix.com/documents/current/api/twisted.internet.ssl.CertificateOptions.html
+    # XXX can we get CertificateOptions to do this for us?
+    from twisted.web import http
+    protocols = [b'http/1.1']
+    if http.H2_ENABLED:
+        protocols = [b'h2', b'http/1.1']
```

Comment:
This shouldn't be necessary: CertificateOptions checks based on the factory that creates the protocol, see `IProtocolNegotiationFactory` and [this code](https://github.com/twisted/twisted/blob/trunk/src/twisted/web/server.py#L787-L797). If you're using a custom protocol factory, you may want to implement `IProtocolNegotiationFactory` instead.